File: 01/python/day1-2.py
#!/usr/bin/env python


all_calories: list[int] = []
cur_calories: int = 0
elf_count: int = 0

with open("../elf_data.txt") as elf_data:

    for line_calories in elf_data:
        if not line_calories.strip():
            all_calories.append(cur_calories)
            print(f"🧝‍♀️ number {len(all_calories)} has {cur_calories} calories.")
            cur_calories = 0
        else:
            cur_calories += int(line_calories)

all_calories.sort(reverse=True)
top3_calories = all_calories[:3]
print()
print(f"The 🧝‍♀️ with the most calories has {top3_calories[0]} calories.")
print(f"The top 3 🧝‍♀️ are carrying {sum(top3_calories)} calories")

File: 01/typescript/day1-2.ts
var lineReader = require('line-reader');

let all_calories: Array<number> = [];
let cur_calories: number = 0;


lineReader.eachLine('../elf_data.txt', function(line: string, last: boolean) {
  let each_ln = line.trim();
  if (each_ln.length == 0) { 
    all_calories.push(cur_calories);
    console.log(`🧝‍♀️ number ${all_calories.length} has ${cur_calories} calories.`);
    cur_calories = 0;
  } else {
    cur_calories = cur_calories + Number(each_ln);
    
  }

  if (last) {
    all_calories.sort((one, two) => (one > two ? -1 : 1));

    console.log(`\nThe 🧝‍♀️ with the most calories has ${all_calories[0]} calories.`);
    console.log(`\nThe top 3 🧝‍♀️ with the most calories have ${all_calories.slice(0,3).reduce((a, b) => a + b, 0)} calories.`)

    return false; // stop reading
    
    141783
  }
});